pkg/apis/hive/v1alpha1/validating-webhooks/clusterdeployment_validating_admission_hook.go

```diff
@@ -607,13 +607,31 @@ func validateMachinePools(cd *hivev1.ClusterDeployment) field.ErrorList {
 
 	vErrs = append(vErrs, validateMachinePool(cd, &cd.Spec.ControlPlane, field.NewPath("spec", "controlPlane"))...)
 
+	vErrs = append(vErrs, validateComputeMachinePoolLength(cd)...)
+
 	for i, mp := range cd.Spec.Compute {
 		vErrs = append(vErrs, validateMachinePool(cd, &mp, field.NewPath("spec", "compute").Index(i))...)
 	}
 
 	return vErrs
 }
 
+func validateComputeMachinePoolLength(cd *hivev1.ClusterDeployment) field.ErrorList {
+	vErrs := field.ErrorList{}
+
+	if cd.Spec.Platform.GCP != nil {
+		// Only restrict the length of compute machine pools on GCP.
```

Comment:
Don't we also need to restrict the name of the machine pool to "worker"?

Reply:
yes, it does appear that the installer forces us to use the name "worker"